routing/router.go

```diff
@@ -1252,6 +1228,107 @@ func (r *ChannelRouter) networkHandler() {
 	}
 }
 
+func (r *ChannelRouter) getMissingBlocks(currentHeight uint32,
+	chainUpdate *chainview.FilteredBlock) {
+
+	outdatedHash, err := r.cfg.Chain.GetBlockHash(int64(currentHeight))
+	if err != nil {
+		log.Error("unable to retrieve outdated block hash")
+	}
+
+	outdatedBlock := &chainntnfs.BlockEpoch{
+		Height: int32(currentHeight),
+		Hash:   outdatedHash,
+	}
+
+	epochClient, err := r.cfg.Notifier.RegisterBlockEpochNtfn(
+		outdatedBlock,
+	)
+	if err != nil {
+		log.Error("unable to register notifications")
+	}
+	defer epochClient.Cancel()
+
+	blockDifference := int(chainUpdate.Height - currentHeight)
+
+	// We'll walk through all the outdated blocks and make sure we're able
+	// to update the graph with any closed channels from them.
+	for i := 0; i < blockDifference; i++ {
+		var missingBlock *chainntnfs.BlockEpoch
+
+		select {
+		case missingBlock = <-epochClient.Epochs:
+		case <-r.quit:
+			return
+		}
+
+		filteredBlock, err := r.cfg.ChainView.FilterBlock(
+			missingBlock.Hash,
+		)
+		if err != nil {
+			log.Errorf("unable to filter block")
+		}
+
+		err = r.updateGraphWithClosedChannels(
+			filteredBlock,
+		)
+		if err != nil {
+			log.Errorf("unable to prune channels: %v", err)
+		}
+	}
+}
+
+func (r *ChannelRouter) updateGraphWithClosedChannels(
+	chainUpdate *chainview.FilteredBlock) error {
+
+	// Once a new block arrives, we update our running
+	// track of the height of the chain tip.
+	blockHeight := chainUpdate.Height
+
+	atomic.StoreUint32(&r.bestHeight, blockHeight)
```

Comment:
This doesn't change behavior as-is, but what if we updated the height only after a successful call to `PruneGraph`? Then the height update of the router is in-sync with the pruning rather than a potential `PruneGraph` failure and the router height is already updated